tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java

```diff
@@ -155,6 +161,94 @@ private void printGroupStates(List<Tuple2<String, String>> groupsAndStates) {
+            ShareGroupDescription description = getDescribeGroups();
+            printGroupDescriptionTable(description);
+        }
+        ShareGroupDescription getDescribeGroups() throws ExecutionException, InterruptedException {
+            String group = opts.options.valueOf(opts.groupOpt);
+            DescribeShareGroupsResult result = adminClient.describeShareGroups(Collections.singletonList(group));
+            Map<String, ShareGroupDescription> descriptionMap = result.all().get();
+            if (descriptionMap.containsKey(group)) {
+                return descriptionMap.get(group);
+            }
+            return null;
+        }
+
+        private void printGroupDescriptionTable(ShareGroupDescription description) {
+            boolean shouldPrintState = opts.options.has(opts.stateOpt);
+            boolean shouldPrintMemDetails = opts.options.has(opts.membersOpt);
+            if (description == null) {
+                return;
+            }
+            if (shouldPrintMemDetails) {
+                printMemberDetails(description.members());
+                return;
+            }
+            List<String> lineItem = new ArrayList<>();
+            lineItem.add(description.groupId());
+            System.out.println(description.members());
+            lineItem.add(description.members().stream().map(MemberDescription::consumerId).collect(Collectors.joining(",")));
+            lineItem.add(description.coordinator().idString());
+            if (shouldPrintState) {
+                lineItem.add(description.state().toString());
+            }
+
+            int maxItemLength = 20;
+            for (String item : lineItem) {
+                if (item != null) {
+                    maxItemLength = Math.max(maxItemLength, item.length());
+                }
+            }
+
+            // header
+            String formatAtom = "%" + (-maxItemLength) + "s";
+            String formatHeader = "";
+            if (shouldPrintState) {
+                formatHeader = String.format(formatAtom + " " + formatAtom + " " + formatAtom + " " + formatAtom, "GROUP_ID", "MEMBERS", "COORDINATOR_NODE", "STATE");
+            } else {
+                formatHeader = String.format(formatAtom + " " + formatAtom + " " + formatAtom, "GROUP_ID", "MEMBERS", "COORDINATOR_NODE");
+            }
+            System.out.println(formatHeader);
+            for (String item : lineItem) {
+                System.out.printf(formatAtom + " ", item);
+            }
+            System.out.println();
+        }
+
+        private void printMemberDetails(Collection<MemberDescription> members) {
+            List<List<String>> lineItems = new ArrayList<>();
+            int maxLen = 20;
+            for (MemberDescription member : members) {
+                List<String> lineItem = new ArrayList<>();
+                lineItem.add(member.consumerId());
+                lineItem.add(member.groupInstanceId().isPresent() ? member.groupInstanceId().get() : "");
```

Comment:
There is no group instance ID for share groups, so this item can be omitted from the output entirely.